### PR TITLE
Convert a few trivial Type, RuntimeType and Enum FCalls to C#

### PR DESCRIPTION
**Pull request by jkotas**

### CODE REVIEW

**Comment by jkotas** [comment]
This makes `Enum.GetUnderlyingType` about 6x faster.

**Review by stephentoub — APPROVED** [review]

**Review by EgorBo — APPROVED** [review]

**Comment by jkotas** [comment]
I am going to revert the change for IsEnum. It is behavior change for generic variables. We do not have a test coverage for it. The test coverage needs to be fixed first.

**Comment by jkotas** [comment]
For example, this prints true currently, but it would print false with this change:
```
Type param = typeof(MyG<>).GetGenericArguments()[0];
Console.WriteLine(param.IsEnum);

class MyG<T> where T: Enum
{
}
```

**Comment by jkotas** [comment]
And this fails with ` Internal CLR error.` currently:
```
Type param = typeof(MyG<>).GetGenericArguments()[0];
Console.WriteLine(Enum.GetUnderlyingType(param));

class MyG<T> where T: Enum
{
}
```

**Comment by jkotas** [comment]
> And this fails with Internal CLR error. currently:

Opened https://github.com/dotnet/runtime/issues/71884

**Review by stephentoub — APPROVED** [review]

File: src/coreclr/System.Private.CoreLib/src/System/Enum.CoreCLR.cs [hunk 1]
```diff
@@ -17,10 +18,59 @@ public abstract partial class Enum
...
+            null,
+            (RuntimeType)typeof(nint),
+            (RuntimeType)typeof(nuint)
+        };
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal static unsafe RuntimeType InternalGetUnderlyingType(RuntimeType enumType)
+        {
+            // Sanity check the last element in the table
+            Debug.Assert(s_underlyingTypes[(int)CorElementType.ELEMENT_TYPE_U] == typeof(nuint));
```
[the remark this answers is not in the document]
Comment (jkotas) [reply src/coreclr/System.Private.CoreLib/src/System/Enum.CoreCLR.cs:66]: I would think we are sufficiently covered between this sanify check assert and the functional tests.

**Comment by dakersnar** [comment]
Improvements on ubuntu arm64: https://github.com/dotnet/perf-autofiling-issues/issues/6982

**Pull request merged**